File: 1001 - 1500/1201-1300/1248. Count Number of Nice Subarrays/freq_arr.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
        int n = nums.length;
        int[] cnt = new int[n + 1];
        cnt[0] = 1;
        int ans = 0, t = 0;
        for (int i = 0; i < n; i++) {
            t += nums[i] & 1;
            if(t-k >= 0) ans+=cnt[t-k];
            cnt[t]++;
        }
        return ans;
    }
}

File: 0001 - 0500/201-300/281-290/287. Find the Duplicate Number/freq_arr.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        vector<int> cnt(nums.size(),0);
        for(int i = 0; i<nums.size(); i++) {
            if(cnt[nums[i]] == 1) return nums[i];
            cnt[nums[i]]++;
        }
        return -1;
    }
};